### DISCUSSION TITLE
Looking for advice on best setup

### DISCUSSION THREAD

**Discussion by Drizzt321** [blakeblackshear/frigate#2177] [Ask A Question]
So I'm running HassOS as a VM on top of FreeBSD bhyve. I have a ZFS RAIDZ2 setup as the storage, and I was thinking I'd just spin up another VM with plenty of storage for saved videos to install Frigate on and test it, before buying a Coral TPU module and using PCIe Passthrough to the VM. Possibly buy a Quadro P400 in the future and pass that through as well for NVENC/NVDEC functionality, but only if I need.

However, from what I'm seeing there's really only Docker image support to install, and there doesn't seem to be much in the way of any kind of installation documentation around 'bare metal' installs. I do see talk of a [Frigate Proxy AddOn](https://www.reddit.com/r/homeassistant/comments/py32ph/question_about_haos_frigate_and_network_storage/) to use Frigate on a separate instance/container, which is what I figure I'll use, as well as install MQTT in HA and use that for the MQTT requirements. 

So, to test out if this is the NVR system I want, how would I best go about a separate instance? Preferably without having to nest docker within that.

**Comment by blakeblackshear** [comment]
Run a debian VM and install docker for running frigate. There is no reason to try and avoid layering docker on the VM. It doesn't introduce any overhead and keeps the OS clean. It also makes upgrades to frigate painless.

**Reply by Drizzt321** [reply]
@blakeblackshear so there must be some extra overhead (I'll believe very minimal), and given it's a single-purpose VM the "keeping the OS clean" is meh, don't care for me.

But I do take your point, and you're right. Kinda the reason I just used the HassOS VM image, rather than installing/configuring it all on top of a bare VM.

**Reply by bobsusedtires** [reply]
> Run a debian VM and install docker for running frigate. There is no reason to try and avoid layering docker on the VM. It doesn't introduce any overhead and keeps the OS clean. It also makes upgrades to frigate painless.

Maybe it's just over my head. I haven't done too much with docker, but that's the best instructions I have seen so far. "install docker for running frigate" Is there anywhere that just gives a step by step?
I have an RPI4 with Coral USB running latest Ubuntu Server, and want to try this instead of Shinobi.

**Reply by blakeblackshear** [reply]
Just follow the standard docker install guide: https://docs.docker.com/engine/install/debian/

Then install compose: https://docs.docker.com/compose/install/

Now you can create a docker compose file and a frigate config file as shown in the docs.

**Reply by veno77** [reply]
@blakeblackshear I have similar observing with coral tpus. When using one coral inference speed is ~7ms, when adding second one the inference speed is ~10ms. Both corals are M.2 - one 2280 and one 2230, cpu stays the same in both cases ~15% utilization on i3-9100(4cores). I have 4 cameras, two 1080p and two 720p. Detection on all is set at their native resolutions. Any ideas why tpu inference goes up? There was a guy who posted here or on fb that with 3 USB Corals he has ~15ms inference per TPU.

**Reply by blakeblackshear** [reply]
It's hard to say. It could be bandwidth constraints on the hardware.

**Reply by grantwest** [reply]
Docker on Linux has zero overhead. Containers are just fancy use of a unix feature called namespaces. The one place that there is technically any overhead is in the networking if you use virtual networks because it introduces another layer of routing, but this will be negligible in all but the most extreme of use cases. Docker does have overhead on Windows and Mac because in order to run linux containers on those platforms there needs to be a linux VM running, thus you get the normal VM overhead on those platforms.
